docs/design/Non_Admin_Controller_design.md

```diff
@@ -0,0 +1,133 @@
+- If the `enableNonAdmin` flag is set to `true`, the OADP Operator will install the NAC in OADP Operator's install namespace, the default value for `enableNonAdmin` will be `false` 
+
+## Pre-requisites
+- **OADP installed**: OADP must be installed and configured to use non-admin controller
+-  **Non Admin Controller configured**: Data Protection Application (DPA) instance must configure Non Admin Controller to watch user namespace(s), by default it watches all Namespaces
+- **RBAC priviledges for the user**: User must have the appropriate RBAC priviledges to create Non Admin Backup object within the Namespace where Backup will be taken. An example of such ClusterRole, which may be added to the user with `RoleBinding`:
+    ```yaml
+    # permissions for end users to edit nonadminbackups.
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRole
+    metadata:
+    labels:
+        app.kubernetes.io/name: clusterrole
+        app.kubernetes.io/instance: nonadminbackup-editor-role
+        app.kubernetes.io/component: rbac
+        app.kubernetes.io/created-by: oadp-nac
+        app.kubernetes.io/part-of: oadp-nac
+        app.kubernetes.io/managed-by: kustomize
+    name: nonadminbackup-editor-role
+    rules:
+    - apiGroups:
+        - nac.oadp.openshift.io
+      resources:
+        - nonadminbackups
+      verbs:
+        - create
+        - delete
+        - get
+        - list
+        - patch
+        - update
+        - watch
+    - apiGroups:
+        - nac.oadp.openshift.io
+      resources:
+        - nonadminbackups/status
+      verbs:
+        - get
+
+    ```
+
+## High-Level design
+
+### Components
+- OADP Operator: OADP is the OpenShift API for Data Protection operator. This open source operator sets up and installs Velero on the OpenShift platform, allowing users to backup and restore applications.
+- Controllers: The Non-Admin controller will pack the following controllers as part of it:
+    - Non-Admin Backup (NAB) Controller: The responsibilities of the NAB controller are:
+        - Validate whether the non-admin user has appropriate administrative namepsace access 
+        - Validate Wehther the non-admin user has appropriate access to create/view/update /delete Non-Admin Backup CR
+        - Listen to requests pertaining to Non-Admin Backup CRD across all the namespaces 
+        - Process requests pertaining to Non-Admin Backup CRD across all the namespaces 
+        - Update Non-Admin CR status with the status/events from Velero Backup CR
+        - Cascade Any actions performed on Non-Admin Backup CR to corresponding Velero backup CR
+    - Non-Admin Restore (NAR) Controller
+- CRDs: The following CRDs will be provided to Non-Admin users:
+    - Non-Admin Backup (NAB) CRD: This iCRD will encapsulate the whole Velero Backup CRD and some additional spec felds that will be needed for non-admin feature.
+    - Non-Admin Restore (NAR) CRD
```

Comment:
Since schedules aren't included here, can we assume that the above mention of schedules is out-of-scope for now? Users who want scheduled NonAdminBackups will need to generate their own CronJobs outside of OADP.